site/src/components/Table.js

```diff
@@ -1,7 +1,56 @@
-import React from 'react';
+import React, { isValidElement } from 'react';
 
-const Table = (props) => <div class="hds-table-container" tabindex="0">
-<table class="hds-table hds-table--dark" aria-label="Service users (dark variant)">{props.children}</table></div>
+const captionPrefix = '[';
+const captionSuffix = ']';
+
+const isCaption = (cellContent) =>
+  typeof cellContent === 'string' && cellContent.startsWith(captionPrefix) && cellContent.endsWith(captionSuffix);
+
+const resolveCaptionStrings = (captionString) => {
+  const withoutSuffixes = captionString.replace(captionPrefix, '').replace(captionSuffix, '');
+  return withoutSuffixes.split(':');
+};
+
+const Table = (props) => {
+  const thead = props.children[0].type.name === 'TableHead' ? props.children[0] : [];
+  const tbody = props.children[1].type.name === 'TableBody' ? props.children[1] : []
+  const tbodyRows = tbody.props?.children || [];
+  const rowsArray = Array.isArray(tbodyRows) ? tbodyRows : [tbodyRows];
+
+  const [rows, captionString] = rowsArray
+    .filter((child) => isValidElement(child))
+    .reduce(
+      (acc, row, i, arr) => {
+        if (
+          i >= arr.length - 1 &&
```

Comment:
`i` can never be > `arr.length -1 `. Small thing, but the whole `if` is quite complicated, so ">=" made me wonder even more what it does.

I don't like using optional chaining (or is it even possible in this environment), but it would make this more readable.

`if ( i=== arr.length - 1 && isCaption(row.props.children?.[0].props.children))`

Reply:
Thanks! Fixed.